### PR TITLE
[11.0] [MIG] Hotel Reservation

### PR DESCRIPTION
**Pull request by serpentcs-dev1**
Hotel Reservation
==================

This Module provides reservation for hotel rooms.

You can manage:

* **Guest Reservation**

* **Group Reservation**

* **Manage Hotel Room Based on Room Types and It's Capacity**

* **Confirm Your Reservation And Get Reservation Information**

* **Generate Folio**

* **Summarize Detail of Reservation**

* **Easy And Quick Reservation from Reservation Summary**

### CODE REVIEW

File: hotel_reservation/static/src/js/hotel_room_summary.js [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+         this.summary_header = py.eval(this.get("summary_header"));
+        }
+        if (this.get("room_summary")) {
+         this.room_summary = py.eval(this.get("room_summary"));
+        }
+        this.renderElement();
+        this.view_loading();
+     },
+     initialize_field: function() {
+         form_common.ReinitializeWidgetMixin.initialize_field.call(this);
```
Comment (vbezl) [inline hotel_reservation/static/src/js/hotel_room_summary.js]: no declaration of form_common

File: hotel_reservation/static/src/js/hotel_room_summary.js [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+            });
+        this.set({
+            date_to: false,
+            date_from: false,
+            summary_header: false,
+            room_summary: false,
+        });
+        this.summary_header = [];
+        this.room_summary = [];
+        this.on("_onFieldChanged:date_from", this, function() {
```
Comment (vbezl) [inline hotel_reservation/static/src/js/hotel_room_summary.js:38]: this doesn't work for some reason - nothing happens when date_from is changed - and nothing is printed to js console...

**Comment by JayVora-SerpentCS** [comment]
@YogeshMahera-SerpentCS @serpentcs-dev1 plz see

File: hotel_reservation/static/src/js/hotel_room_summary.js (hunk 2 above)
Comment (YogeshMahera-SerpentCS) [reply hotel_reservation/static/src/js/hotel_room_summary.js:38]: @vbezl I have checked in local it's working fine. I think once runbot will start you can test it.

**Comment by YogeshMahera-SerpentCS** [comment]
@omarouhari @gustavovalverde @DarioLodeiros  Can you please Code/Functional Review this, so we could merge this and move ahead.

File: hotel_reservation/__manifest__.py [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+    'version': '11.0.1.0.0',
+    'author': 'Odoo Community Association (OCA), Serpent Consulting \
+                Services Pvt. Ltd., Odoo S.A.',
+    'category': 'Generic Modules/Hotel Reservation',
+    'license': 'AGPL-3',
+    'summary': 'Manages Guest Reservation & displays Reservation Summary',
+    'depends': ['hotel', 'stock', 'mail'],
+    'demo': ['views/hotel_reservation_data.xml', ],
+    'data': [
+        'security/ir.model.access.csv',
```
Comment (nikul-serpentcs) [inline hotel_reservation/__manifest__.py:14]: @serpentcs-dev1 Use Sequence

File: hotel_reservation/__manifest__.py [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+{
+    'name': 'Hotel Reservation Management',
+    'version': '11.0.1.0.0',
+    'author': 'Odoo Community Association (OCA), Serpent Consulting \
+                Services Pvt. Ltd., Odoo S.A.',
+    'category': 'Generic Modules/Hotel Reservation',
+    'license': 'AGPL-3',
+    'summary': 'Manages Guest Reservation & displays Reservation Summary',
+    'depends': ['hotel', 'stock', 'mail'],
+    'demo': ['views/hotel_reservation_data.xml', ],
```
Comment (nikul-serpentcs) [inline hotel_reservation/__manifest__.py]: demo data always `demo`  folder or directory

File: hotel_reservation/models/hotel_reservation.py [hunk 5]
```diff
@@ -0,0 +1,1064 @@
...
+        -----------------------------------------------------
+        @param self: object pointer
+        @return: raise a warning depending on the validation
+        '''
+        ctx = dict(self._context) or {}
+        for reservation in self:
+            cap = 0
+            for rec in reservation.reservation_line:
+                if len(rec.reserve) == 0:
+                    raise ValidationError(_('Please Select Rooms \
```
Comment (nikul-serpentcs) [inline hotel_reservation/models/hotel_reservation.py]: add Doc string

File: hotel_reservation/views/room_summ_view.xml [hunk 6]
```diff
@@ -0,0 +1,16 @@
...
+<!DOCTYPE xml>
+<odoo>
+
+    <template id="assets_backend_" name="Quick Reservation Summary"
+        inherit_id="web.assets_backend">
+        <xpath expr="." position="inside">
+            <link rel="stylesheet" href="/hotel_reservation/static/src/css/room_summary.css" />
+            <script type="text/javascript"
+                src="/hotel_reservation/static/src/js/hotel_room_summary.js"></script>
+            <!-- <script type="text/javascript"
```
Comment (nikul-serpentcs) [inline hotel_reservation/views/room_summ_view.xml]: remove dead code

**Review by nikul-serpentcs — COMMENTED** [review]
Improve Code

**Comment by YogeshMahera-SerpentCS** [comment]
@nikul-serpentcs Amended Changes!

File: hotel_reservation/report/hotel_reservation_report.py [hunk 7]
```diff
@@ -0,0 +1,231 @@
...
+
+class ReportTestCheckin(models.AbstractModel):
+    _name = "report.hotel_reservation.report_checkin_qweb"
+
+    def _get_room_type(self, date_start, date_end):
+        reservation_obj = self.env['hotel.reservation']
+        room_dom = [('checkin', '>=', date_start),
+                    ('checkout', '<=', date_end)]
+        tids = reservation_obj.search(room_dom)
+        res = reservation_obj.browse(tids)
```
Comment (luistorresm) [inline hotel_reservation/report/hotel_reservation_report.py]: I think that `tids ` already is a browse object.

File: hotel_reservation/report/hotel_reservation_report.py [hunk 8]
```diff
@@ -0,0 +1,231 @@
...
+        return res
+
+    def _get_room_nos(self, date_start, date_end):
+        reservation_obj = self.env['hotel.reservation']
+        tids = reservation_obj.search([('checkin', '>=', date_start),
+                                       ('checkout', '<=', date_end)])
+        res = reservation_obj.browse(tids)
```
Comment (luistorresm) [inline hotel_reservation/report/hotel_reservation_report.py]: The same here

Comment (JayVora-SerpentCS) [reply hotel_reservation/report/hotel_reservation_report.py]: @YogeshMahera-SerpentCS do check

File: hotel_reservation/README.rst [hunk 9]
```diff
@@ -0,0 +1,83 @@
+.. image:: https://img.shields.io/badge/licence-AGPL--3-blue.svg
+   :target: https://www.gnu.org/licenses/agpl
+   :alt: License: AGPL-3
+
+==================
+Hotel Reservation
+==================
+
+This Module is provide reservation for hotel rooms.
```
Comment (JayVora-SerpentCS) [inline hotel_reservation/README.rst]: This module provides the feature of Reserving the Hotel Rooms.

File: hotel_reservation/__manifest__.py (hunk 3 above)
Comment (JayVora-SerpentCS) [reply hotel_reservation/__manifest__.py:14]: use the readme generator : https://github.com/OCA/maintainer-tools/blob/master/tools/gen_addon_readme.py

Comment (JayVora-SerpentCS) [reply hotel_reservation/__manifest__.py:14]: Follow utmost : https://github.com/OCA/maintainer-tools/tree/master/template/module

File: hotel_reservation/data/hotel_reservation_data.xml [hunk 10]
```diff
@@ -0,0 +1,24 @@
+<?xml version="1.0" encoding="utf-8"?>
+<!DOCTYPE xml>
+<odoo noupdate="1">
+
+    <!-- Demo data for hotel reservation -->
+    <record id="hotel_reservation_0" model="hotel.reservation">
+        <field eval="&quot;2013-05-02 10:00:00&quot;" name="date_order" />
+        <field eval="&quot;2013-05-05 12:09:00&quot;" name="checkout" />
+        <field eval="&quot;2013-05-02 12:09:00&quot;" name="checkin" />
```
Comment (JayVora-SerpentCS) [inline hotel_reservation/data/hotel_reservation_data.xml]: need 2018 dates now:)

**Review by JayVora-SerpentCS — CHANGES_REQUESTED** [review]

File: hotel_reservation/__manifest__.py (hunk 3 above)
Comment (nikul-serpentcs) [reply hotel_reservation/__manifest__.py:14]: @JayVora-SerpentCS He is already followed [this](https://github.com/OCA/maintainer-tools/blob/master/template/module/__manifest__.py#L27)

**Comment by YogeshMahera-SerpentCS** [comment]
@luistorresm @JayVora-SerpentCS Amended Changes!

**Comment by JayVora-SerpentCS** [comment]
squash plz

**Comment by YogeshMahera-SerpentCS** [comment]
@JayVora-SerpentCS Squash done

**Pull request merged**